src/main/java/fr/sncf/osrd/interactive/InteractiveSimulation.java

```diff
@@ -165,4 +164,38 @@ public void sendTrainDelays(Collection<String> trains) throws IOException {
         }
         sendResponse(new ServerMessage.TrainDelays(trainDelays));
     }
+
+    /** Sends back to the client the train succession tables.
+     * If waiting for simulation then return the completed TST of last simulation.
+     * @param switches only return tst of the given switches. If `null` or empty then return the whole list.
+     */
+    public void sendTrainSuccessionTables(List<String> switches) throws IOException {
+        if (expectState(SessionState.PAUSED, SessionState.WAITING_FOR_SIMULATION))
+            return;
+        if (simulation == null) {
+            var details = new TreeMap<String, String>();
+            details.put("reason", "waiting for the first simulation to be run");
+            sendResponse(new ServerMessage.Error("Can't send train succession tables", details));
+            return;
+        }
```

Comment:
we should add a SIMULATION_COMPLETE state, so that we don't have to check for that